test/E2ETest/Microsoft.Test.E2E.AspNet.OData/QueryComposition/OrderByTests.cs

```diff
@@ -107,6 +107,20 @@ public async Task CanOrderByDuplicatePropertiesSimiliarPath()
                             previousElement.WorkAddress.CountryOrRegion.Name.CompareTo(currentElement.WorkAddress.CountryOrRegion.Name) < 1);
             }
         }
+
+
+        [Fact]
+        public async Task CanOrderByDerivedComplexTypeProperties()
+        {
+            string query =
+                "/odata/OrderByCustomers?$orderby=Address/Microsoft.Test.E2E.AspNet.OData.QueryComposition.CustomOrderByAddress/CustomFirstLine asc";
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, BaseAddress + query);
+            HttpResponseMessage response = await Client.SendAsync(request);
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            dynamic parsedContent = JObject.Parse(await response.Content.ReadAsStringAsync());
+            Assert.NotNull(parsedContent.value);
```

Comment:
@KenitoInc What this test doesn't do is verify the result of sorting by the particular property on the derived type. See in a different test how someone went ahead to assert the result or ordering by a particular property: https://github.com/OData/WebApi/pull/2733/files#diff-6def845c9ce5041798d91cdbcab7285a1a0c6cdac96eb59bbfc0b3fc60104545R96-R107